vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/model/implementation/Serializer.java

```diff
@@ -0,0 +1,286 @@
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import org.semanticweb.vlog4j.core.model.api.Conjunction;
+import org.semanticweb.vlog4j.core.model.api.Constant;
+import org.semanticweb.vlog4j.core.model.api.DataSourceDeclaration;
+import org.semanticweb.vlog4j.core.model.api.DatatypeConstant;
+import org.semanticweb.vlog4j.core.model.api.ExistentialVariable;
+import org.semanticweb.vlog4j.core.model.api.Fact;
+import org.semanticweb.vlog4j.core.model.api.LanguageStringConstant;
+import org.semanticweb.vlog4j.core.model.api.Literal;
+import org.semanticweb.vlog4j.core.model.api.NamedNull;
+import org.semanticweb.vlog4j.core.model.api.Predicate;
+import org.semanticweb.vlog4j.core.model.api.PrefixDeclarations;
+import org.semanticweb.vlog4j.core.model.api.Rule;
+import org.semanticweb.vlog4j.core.model.api.Term;
+import org.semanticweb.vlog4j.core.model.api.UniversalVariable;
+
+/**
+ * A utility class with static methods to obtain the correct parsable string
+ * representation of the different data models.
+ * 
+ * @author Ali Elhalawati
+ *
+ */
+public final class Serializer {
+	public static final String NEGATIVE_IDENTIFIER = "~";
+	public static final String COMMA = ", ";
+	public static final String DOT = ".";
+	public static final String EXISTENTIAL_IDENTIFIER = "!";
+	public static final String UNIVERSAL_IDENTIFIER = "?";
+	public static final String NAMEDNULL_IDENTIFIER = "_";
+	public static final String OPEN_PARENTHESIS = "(";
+	public static final String CLOSING_PARENTHESIS = ")";
+	public static final String RULE_SEPARATOR = " :- ";
+	public static final String AT = "@";
+	public static final String SOURCE = "@source ";
+	public static final String COLON = ": ";
+	public static final String COLON_UNSPACED = ":";
+	public static final String CARET = "^";
+	public static final String LESS_THAN = "<";
+	public static final String MORE_THAN = ">";
+	public static final String QUOTE = "\"";
+	public static final String DOUBLE = "^[-+]?[0-9]+[.]?[0-9]*([eE][-+]?[0-9]+)?$";
+	public static final String INTEGER = "^[-+]?\\d+$";
+	public static final String DECIMAL = "^(\\d*\\.)?\\d+$";
+	public static final String TRUE = "true";
+	public static final String FALSE = "false";
+
+	/**
+	 * Constructor.
+	 */
+	private Serializer() {
+
+	}
+
+	private static String escape(String string) {
+		return string.replace("\\", "\\\\").replace("\"", "\\\"");
+	}
+
+	private static String checkRelativeAbsoluteIri(String string) {
+		if ((string.contains(COLON_UNSPACED) || string.matches(INTEGER) || string.matches(DOUBLE)
+				|| string.matches(DECIMAL) || string.equals(TRUE) || string.equals(FALSE))
+				&& (!string.contains(LESS_THAN))) {
```

Comment:
because we want to make sure that this doesn't have < > already

Reply:
but it only checks for "<". So if you have "bla>", the toString will be "<bla>>".
Anyway, a "parsed" string that represents an IRI with never create an AbstractConstant or Predicate with name containing "<" or ">".  The parser does not decode "%3C", which is the URL encoding of "<", into "<".
In the future, if the IRI contains special predicates (like "<", for example), in the toString we would have to encode it (for example, replace it with "%3C"). This can be done in another task.